Work for Linked Lists

Lnleite opened the pull request:
# Tasks
- [x] Queues
  - [x] enqueue
  - [x] dequeue
  - [x] len
- [x] Binary Search Trees
  - [x] insert
  - [x] contains
  - [x] get_max

- [x] Doubly Linked Lists
  - [x] add_to_head
  - [x] add_to_tail
  - [x] remove_from_head
  - [x] remove_from_tail
  - [x] move_to_front
  - [x] move_to_end
  - [x] delete
  - [x] get_max
- [x] Heaps
  - [x] insert
  - [x] delete
  - [x] get_max
  - [x] get_size
  - [x] _bubble_up
  - [x] _sift_down.

Lnleite committed:
Work for Linked Lists

Lnleite committed:
Fix remove_head, complete get_max

The remove_head was not finished from the lecture because we did not change the tail
when we had a None value in the head.

Lnleite committed:
Fix == and != to is and is not

Lnleite committed:
Finish queue

No struggles, pretty straight forward.

Lnleite committed:
Change .value to get_value()

Lnleite committed:
Add tail and head functions work, get_max almost there

Lnleite committed:
get_max functional

Lnleite committed:
Solve double_link_list functions

I was able to now figure out the functions after getting some sleep.
I realized I over complicated the functions and actually need to let the Node's
functions do the work.

Lnleite committed:
Finish doubly_linked_list

After I understood the node functions it was all down hill from there. Finished the
move_to_front and move_to_end functions.

Lnleite committed:
Add delete test and fix delete function

Lnleite committed:
Finish insert

Lnleite committed:
Finish insert for binary search tree

Lnleite committed:
Finish contains for binary search tree

Lnleite committed:
Finish get_max for binary search tree

Lnleite committed:
Answer DS questions, everything except Heaps

AI2EJAY commented on binary_search_tree/binary_search_tree.py:
```diff
@@ -37,7 +37,14 @@ def rec(value, current_node):
 
     def get_max(self):
-        pass
+        maxValue = -float("inf")
```
Nice job incorporating what we learned in the buying_stocks problem we had in algo's

AI2EJAY commented on binary_search_tree/binary_search_tree.py:
```diff
@@ -1,14 +1,58 @@
+                    current_node.left = node
+                    return
+                else:
+                    rec(target, current_node.left)
```
Nice recursive approach!

AI2EJAY commented on queue/queue.py:
```diff
@@ -1,17 +1,22 @@
 import sys
 sys.path.append('../linked_list')
-from linked_list import LinkedList
+from linked_list import LinkedList  # noqa
```
Thank you for finding and sharing this.

Lnleite committed:
Finish insert, _bubble_up, get_max, and get_size

Had to take a peak at some code to get the buble up. I had the right idea, but I forgot
to subtract 1 from the index.

Lnleite committed:
Finish heap

Got the delete and _sift_down working

Lnleite committed:
Finish data questions
